codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -272,6 +275,12 @@ private boolean readClientPrefaceString(ByteBuf in) throws Http2Exception {
             if (bytesRead == 0 || !ByteBufUtil.equals(in, in.readerIndex(),
                                                       clientPrefaceString, clientPrefaceString.readerIndex(),
                                                       bytesRead)) {
+                String http1Request = in.toString(0, 1024, CharsetUtil.US_ASCII);
```

Comment:
Hmm, I assumed that the size was a maximum, rather than an actual length.  I picked 1024 because 512 is too short, and 2048 too long.